### DISCUSSION TITLE
Error with conf and slow atc

### DISCUSSION THREAD

**Discussion by arkodeepg** [EvgeniyMukhamedjanov/liquid-ajax-cart#91] [Q&A]
Hi Evgeniy I'm facning a bit of issue with conf 
![image](https://github.com/EvgeniyMukhamedjanov/liquid-ajax-cart/assets/43224169/7a0fc3ff-1d2f-4a29-b5b2-7cb9b4a62fd1)
error statement - **Uncaught TypeError: Cannot read properties of undefined (reading 'conf')**

I tried putting it in square brackets too like in the example but sitll getting this error. 

Also facning a bit of issue of the atc taking about 3-5 sec, is there any way of making it faster?
testing link w/ product - https://j0n0jh8gefp2pmtu-26900106.shopifypreview.com/products/blue-flower-ceramic-22-piece-dinner-set-for-6

**Comment by EvgeniyMukhamedjanov** [comment]
Hi @arkodeepg !
The ```Uncaught TypeError: Cannot read properties of undefined (reading 'conf')``` error might happen only if you are trying to read ```window.liquidAjaxCart``` object *before* it had been added to the page. Means your ```global-custom.min.js``` script runs *before* the ```liquid-ajax-cart-v2.1.0-min.js``` file had been imported. And it makes sense, because script modules (where you import liquid ajax cart) are always deferred, meanwhile your ```global-custom.min.js``` is not.

There are a few ways how you can fix that:
* Don't preload ```global-custom.min.js```: instead make this script deferred as well, so it will be run after liquid-ajax-cart.
* If your ```global-custom.min.js``` is a bundle that compiled using webpack, then I would suggest you to import liquid-ajax-cart from npm.js right into ```global-custom.js```.
* Move the liquid-ajax-cart configuration right into the script where you import liquid-ajax-cart and place it under the import.

Regarding "Add to cart" speed. It seems to be slow because liquid-ajax-cart makes two requests when a user adds something to the cart: ```cart/add.js``` + ```cart/update.js```. The update is required because the ```add.js``` doesn't return the full information about the updated cart state and in some cases it returns even wrong cart HTML (issue: https://github.com/EvgeniyMukhamedjanov/liquid-ajax-cart/issues/72). 
But when your theme is live, it should be a bit faster.

Also it is a good point about the speed of "Add to cart". I'll take a look at it one more time if it is possible to somehow make it visually faster, but it is not a quick fix for sure. If I find a reliable solution, I'll definitely let you know :-)

**Comment by arkodeepg** [comment]
![image](https://github.com/EvgeniyMukhamedjanov/liquid-ajax-cart/assets/43224169/262e166f-162c-4910-a430-b4b5d168f394)
there is one more cart.js call hapenning which im not able to find

**Reply by EvgeniyMukhamedjanov** [reply]
It might be an app, that intercepts all ```add.js``` requests for tracking or other purposes. To make sure that it doesn't come from liquid-ajax-cart -- try to send an ```add.js``` request manually from the browser console using the plain ```fetch``` function. Most likely you'll see another ```cart.js``` request as well.

**Comment by arkodeepg** [comment]
![image](https://github.com/EvgeniyMukhamedjanov/liquid-ajax-cart/assets/43224169/cdf6992d-3ef8-4ddb-b0a6-f7e640f126ee)
as for the mutation i've added it like this which mostly works but still sometimes giving error

**Reply by EvgeniyMukhamedjanov** [reply]
You still rely on the ```myMutation``` object that comes from the ```global-custom.min.js``` script that might be loaded after the liquid-ajax-cart library, so the ```myMutation``` might not exist.
